### PR TITLE
feat: add new user

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/UserManagement/InviteUsers.res [hunk 1]
```diff
@@ -112,8 +130,21 @@ let make = () => {
         ]
         ->Dict.fromArray
         ->Js.Json.object_
-      let _ = inviteUserReq(body, index)
+      let _ = inviteUserReq(body, index, emailPasswordsArray)
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/UserManagement/InviteUsers.res:133]: Can we directly pass empty here in the same line?

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/UserManagement/InviteUsers.res:144]: We need array reference name as this variable being used later.

**Review by jainlokesh318 — DISMISSED** [review]

File: src/screens/HyperSwitch/UserManagement/InviteUsers.res [hunk 2]
```diff
@@ -74,47 +75,91 @@ let make = () => {
...
+      valDict
+      ->getStrArray("emailList")
+      ->Array.map(ele => {
+        let body =
+          [
+            ("email", ele->String.toLowerCase->Js.Json.string),
+            ("name", ele->getNameFromEmail->Js.Json.string),
+            ("role_id", role->Js.Json.string),
+          ]
+          ->Dict.fromArray
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/UserManagement/InviteUsers.res:135]: use LogciUtils.getJsonFromArrayOfJson

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/UserManagement/InviteUsers.res]: Resolved

File: src/screens/HyperSwitch/UserManagement/InviteUsers.res [hunk 3]
```diff
@@ -74,32 +75,69 @@ let make = () => {
...
+            (
+              "email",
+              body
+              ->LogicUtils.getDictFromJsonObject
+              ->LogicUtils.getString("email", "")
+              ->Js.Json.string,
+            ),
+            ("password", passwordFromResponse->Js.Json.string),
+          ]
+          ->Dict.fromArray
```
[the remark this answers is not in the document]
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/UserManagement/InviteUsers.res]: Resolved

**Review by Riddhiagrawal001 — APPROVED** [review]

File: src/utils/PromiseUtils.res [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// Promise.allSettled() takes an iterable of promises and returns a single promise that is fulfilled with an array of promise settlement result
+
+let allSettledPolyfill = (arr: array<promise<Js.Json.t>>) => {
+  let res = arr->Array.map(promise =>
+    promise
+    ->Promise.then(val => {
+      Promise.resolve(val)
+    })
+    ->Promise.catch(_ => {
+      Js.Json.null->Js.Promise.resolve
```
Comment (jainlokesh318) [inline src/utils/PromiseUtils.res:11]: Send the error itself

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**